File: starrysky/src/main/java/com/lzx/starrysky/BaseMediaInfo.java
package com.lzx.starrysky;

import android.support.annotation.Nullable;

public class BaseMediaInfo implements Cloneable {
    private String mediaId = "";
    private String mediaUrl;
    private String mediaCover;
    private String mediaTitle;
    private long duration;

    public String getMediaId() {
        return mediaId;
    }

    public void setMediaId(String mediaId) {
        this.mediaId = mediaId;
    }

    public String getMediaUrl() {
        return mediaUrl;
    }

    public void setMediaUrl(String mediaUrl) {
        this.mediaUrl = mediaUrl;
    }

    public String getMediaCover() {
        return mediaCover;
    }

    public void setMediaCover(String mediaCover) {
        this.mediaCover = mediaCover;
    }

    public String getMediaTitle() {
        return mediaTitle;
    }

    public void setMediaTitle(String mediaTitle) {
        this.mediaTitle = mediaTitle;
    }

    public long getDuration() {
        return duration;
    }

    public void setDuration(long duration) {
        this.duration = duration;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj instanceof BaseMediaInfo) {
            BaseMediaInfo info = (BaseMediaInfo) obj;
            return info.getMediaId().equals(mediaId) && info.getMediaUrl().equals(mediaUrl);
        } else {
            return false;
        }
    }

    @Override
    protected Object clone() {
        Object obj = null;
        //调用Object类的clone方法，返回一个Object实例
        try {
            obj = super.clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
        }
        return obj;
    }

    @Override
    public String toString() {
        return "BaseMediaInfo{" +
                "mediaId='" + mediaId + '\'' +
                '}';
    }
}

File: starrysky/src/main/java/com/lzx/starrysky/BaseMediaInfo.kt
package com.lzx.starrysky

class BaseMediaInfo : Cloneable {
    var mediaId = ""
    var mediaUrl: String? = null
    var mediaCover: String? = null
    var mediaTitle: String? = null
    var duration: Long = 0
    var mapHeadData: Map<String, String>? = null

    override fun equals(obj: Any?): Boolean {
        return if (obj is BaseMediaInfo) {
            obj.mediaId == mediaId && obj.mediaUrl == mediaUrl
        } else {
            false
        }
    }

    override fun clone(): Any {
        var obj: Any? = null
        //调用Object类的clone方法，返回一个Object实例
        try {
            obj = super.clone()
        } catch (e: CloneNotSupportedException) {
            e.printStackTrace()
        }
        return obj!!
    }

    override fun toString(): String {
        return "BaseMediaInfo{" +
            "mediaId='" + mediaId + '\'' +
            '}'
    }

    override fun hashCode(): Int {
        var result = mediaId.hashCode()
        result = 31 * result + (mediaUrl?.hashCode() ?: 0)
        result = 31 * result + (mediaCover?.hashCode() ?: 0)
        result = 31 * result + (mediaTitle?.hashCode() ?: 0)
        result = 31 * result + duration.hashCode()
        result = 31 * result + (mapHeadData?.hashCode() ?: 0)
        return result
    }
}